### ISSUE TITLE
How to get sender email address in Teams

### ISSUE THREAD

**Issue by Polkien** [Microsoft/BotFramework-Hubot#26]
Hello! Noob question: Is there a way to get the sender email address when using MS Teams channel?

**Comment by MattSFT** [comment]
Yes, but I'm not sure that this adapter has the correct plumbing to do so. I think we made these changes before Teams provided the email address.

**Comment by MattSFT** [comment]
Right here: https://github.com/Microsoft/BotFramework-Hubot/blob/8d0cf959ff651f49cb4454c26cae968bd092f3a0/src/msteams-middleware.coffee#L110

I believe one of the values that the user object has now is an email address, some name qualities, and the AAD object id

**Comment by Polkien** [comment]
Unfortunately, the email address is not one of the properties of the `activity?.address?.user` object. The object is:

```
{ id: 'xxxxxx',
  name: 'polkien',
  aadObjectId: 'xxxxxx' }
```

Any other idea? 🤔

**Comment by MattSFT** [comment]
The only other option would be to call the roster api upon message receipt:
https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/bots/bots-context#fetching-the-team-roster

**Comment by MattSFT** [comment]
Hey @Polkien were you able to make progress using the team roster API?

**Comment by Polkien** [comment]
@MattSFT, Sorry but the documentation I found for roster api is for creating teams apps (not sure how to integrate with this Hubot adapter). I need time to study everything and find a way to integrate these two pieces. Unfortunately I don't have enough time to spend on this right now but anyone else who has an idea or a clue is welcome to help! (please jeje).

**Comment by MattSFT** [comment]
@Polkien yes this would be teams specific since other channels might not have a notion of email for a user.

**Comment by casieo** [comment]
Hi, just checking to see if there are any updates on this, or possibly an ETA?  It'd be great to be able to grab email address of people using Teams to communicate with bots.

**Comment by vkapur** [comment]
Did anyone resolve this issue?

**Comment by MattSFT** [comment]
@vkapur I do not believe so.

**Comment by flologicshilpa** [comment]
Any updates in this topic? I need this feature as well.

**Comment by MattSFT** [comment]
You will need to call the roster to get the email address. It isn't available at the time of receiving a message unfortunately

**Comment by basxib** [comment]
Do you have a reference for how to call the roster for the V4 of the SDK? I could only find reference for V3 which is incompatible with V4.

**Comment by MattSFT** [comment]
This is how the BotBuilder SDK does it:
https://github.com/OfficeDev/BotBuilder-MicrosoftTeams-node/blob/922d14e660cf5c5b9e510617b4cd01084483593d/samples/teams-bot/src/bot.ts#L63

**Comment by basxib** [comment]
@MattSFT I tried doing that but I'm getting an error when calling either getActivityMembers or getConversationMembers,, the exception is "The bot is not part of the conversation roster."

**Comment by MattSFT** [comment]
That usually happens if you are asking in a conversation where the bot has not been installed.

**Comment by MattSFT** [comment]
Are you testing using sideloading? https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/apps/apps-upload

**Comment by basxib** [comment]
Yes

**Comment by AkshatTR** [comment]
Is there any solution for using TeamsInfo with sideloading apps?

**Comment by MuriloTSGit** [comment]
Hello folks,

I'm trying to get user infos (email, name), anyone knows how can i do this and can explain to me in noob mode ? Thank you

**Comment by tithu** [comment]
Any update on this issue? How can I retrieve email id of logged in user in teams in v3?

**Comment by AkshatTR** [comment]
You can get email of the logged in user with TeamsInfo class but the catch is you need to create a bot even if you only want a messaging extension. Crazy right?

**Comment by tithu** [comment]
I guess TeamsInfo class is supported only in v4 SDK. Currently I have configured my bot in MS teams for v3 SDK.  I don't want only messaging extension. I want users email id(actually teams login id) and want to use that for other purpose. Is there any way I can get email id of current user in BOT V3 SDK?
